mantidproject/mantid
# Sans gui beam center finder

Description of work.
This work implements the BeamCentreFinder command for ISIS sans using the new backend. 
**To test:**
To test run the following two scripts in mantid and check that they produce the same results. 

from sans.command_interface.ISISCommandInterface import *
Clean()
MASKFILE = FileFinder.getFullPath('MaskSANS2DReductionGUI.txt')
BATCHFILE = FileFinder.getFullPath('sans2d_reduction_gui_batch.csv')
UseCompatibilityMode()
SANS2D()
MaskFile(r'C:\Users\tgz88916\Documents\MantidBuild\ExternalData\Testing\Data\SystemTest\SANS2D\MaskSANS2DReductionGUI.txt')
AssignSample('22048')
AssignCan('22023')
TransmissionSample('22041', '22024')
TransmissionCan('22024', '22024')
centre = FindBeamCentre(rlow=41.0, rupp=280.0, xstart=float(150)/1000., ystart=float(-160)/1000., tolerance=0.0001251, MaxIter=3, reduction_method=True)

and

from ISISCommandInterface import *
Clean()
MASKFILE = FileFinder.getFullPath('MaskSANS2DReductionGUI.txt')
BATCHFILE = FileFinder.getFullPath('sans2d_reduction_gui_batch.csv')
SANS2D()
MaskFile(r'C:\Users\tgz88916\Documents\MantidBuild\ExternalData\Testing\Data\SystemTest\SANS2D\MaskSANS2DReductionGUI.txt')
AssignSample('22048')
AssignCan('22023')
TransmissionSample('22041', '22024')
TransmissionCan('22024', '22024')
centre = FindBeamCentre(rlow=41.0, rupp=280.0, xstart=float(150)/1000., ystart=float(-160)/1000., tolerance=0.0001251, MaxIter=3)

The results are shown in the logging window. IN particular check that all the numbers after 3 iterations are the same. 
Fixes #20822 . 

**Release Notes** 
Have added to release notes

Also need to update here https://www.mantidproject.org/Scripting_SANS_Reductions_With_The_New_Reduction_Backend
once merged.


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
